ocaml/xapi/export.ml

```diff
@@ -328,7 +341,39 @@ let make_gpu_group table __context self =
 		snapshot = API.Legacy.To.gPU_group_t group
 	}
 
-let make_all ~with_snapshot_metadata ~preserve_power_state table __context = 
+let make_pvs_proxies table __context self =
+	debug "exporting PVS Proxy %s" (Ref.string_of self);
+	let lookup' ref = lookup table (Ref.string_of ref) in
+	let proxy = Db.PVS_proxy.get_record ~__context ~self in
+	let proxy =
+		{ proxy with
+		  API.pVS_proxy_site      = lookup' proxy.API.pVS_proxy_site
+		; API.pVS_proxy_VIF       = lookup' proxy.API.pVS_proxy_VIF
+		; API.pVS_proxy_cache_SR  = lookup' proxy.API.pVS_proxy_cache_SR
+		} in
+	{ cls      = Datamodel._pvs_proxy
+	; id       = Ref.string_of (lookup' self)
+	; snapshot = API.Legacy.To.pVS_proxy_t proxy
+	}
+
+let make_pvs_sites table __context self =
+	debug "exporting PVS Site %s" (Ref.string_of self);
+	let lookup'  ref  = lookup table (Ref.string_of ref) in
+	let filter'  refs = filter table (List.map Ref.string_of refs) in
+	let site = Db.PVS_site.get_record ~__context ~self in
+	let site =
+		{ site with
+		  API.pVS_site_cache_storage = filter' site.API.pVS_site_cache_storage
+		; API.pVS_site_servers       = filter' site.API.pVS_site_servers
+		; API.pVS_site_proxies       = filter' site.API.pVS_site_proxies
```

Comment:
This last one is important, but isn't it true that the servers are not included and might as well be set to `[]` immediately? And similarly, as above, the cache storage is likely to be different on the receiving pool, and the SRs may not be included in the exported metadata, so this can be set to `[]` as well.